db/schema.rb

```diff
@@ -157,9 +166,12 @@
   end
 
   add_foreign_key "delivery_attempts", "emails", on_delete: :cascade
+  add_foreign_key "digest_run_subscribers", "digest_runs"
+  add_foreign_key "digest_run_subscribers", "subscribers"
```

Comment:
I've just been having a chat with @thomasleese about how these should be set up cascade wise (and this was what I was going to chat to you about it so it might be easier to pick up over a hangout).

So we think that it should be possible to delete DigestRun rows and without them DigestRunSubscriber no longer makes sense so they should be cascaded, and the corresponding subscription_contents should also be removed as they don't make sense without the digest run. Which would indicate: `add_foreign_key "digest_run_subscribers", "digest_runs"` should have an `on_delete: :cascade`

I'd guess we also want to remove digest_run_subscriber entry if the subscriber is removed which would be another `on_delete: :cascade`

Also looks like the email `on_delete: :nullify` is wrong (not your PR of course) as null email value means something. I think resolving this though should be part of working out an email archiving strategy.

Reply:
ok... I think we're good now.